src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ForegroundDispatcher.cs

```diff
@@ -11,10 +12,18 @@ internal abstract class ForegroundDispatcher
     {
         public abstract bool IsForegroundThread { get; }
 
+        public abstract bool IsBackgroundThread { get; }
+
         public abstract TaskScheduler ForegroundScheduler { get; }
 
         public abstract TaskScheduler BackgroundScheduler { get; }
 
+        public Task RunOnForegroundAsync(Action action, CancellationToken cancellationToken)
+            => RunOnForegroundAsync(new Func<Task>(() => { action(); return Task.CompletedTask; }), cancellationToken);
```

Comment:
I would just call Task.Factory.StartNew separately instead of trying to utilize a Func. The Task.Factory.STartNew method has an overload that takes an Action is why